### ISSUE TITLE
/tmp/ files not generated

### ISSUE THREAD

**Issue by wasabigeek** [jbarlow83/OCRmyPDF#169]
I'm getting an error when running `ocrmypdf input.pdf output.pdf`, based on the stacktrace it seems that the `/tmp/com.github.ocrmypdf...` files are not created.

I tried running `mkdtemp()` based on [source](https://github.com/jbarlow83/OCRmyPDF/blob/7f3fa46a40f61f1e78071bc48b108ef4cf13b198/ocrmypdf/__main__.py#L636) in the Python shell. The `/tmp/com.github.ocrmypdf...` folder is created properly.

However, when I run the command, no folders are created. Instead I get the following error:
```
ERROR - Unexpected ruffus exception:

Original exception:

    Exception #1
      '<class 'ruffus.ruffus_exceptions.RethrownJobError'>

        Exceptions generating parameters for

        task = 'ocrmypdf.pipeline.render_hocr_page'

    Original exception:

        Exception #1
          'ruffus.ruffus_exceptions.MissingInputFileError(


            No way to run job: Input file '/tmp/com.github.ocrmypdf.plmusa3s/000001.txt' does not exist)' raised in ...
           Task = def ocrmypdf.pipeline.render_hocr_page(...):


        Traceback (most recent call last):
          File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 5118, in parameter_generator
            job_history, verbose_abbreviated_path):
          File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 4945, in job_needs_to_run
            check_input_files_exist(*params)
          File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/file_name_parameters.py", line 399, in check_input_files_exist
            "Input file '%s' does not exist" % f)
        ruffus.ruffus_exceptions.MissingInputFileError:


            No way to run job: Input file '/tmp/com.github.ocrmypdf.plmusa3s/000001.txt' does not exist

        ' raised in ...
    Traceback (most recent call last):
      File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 5118, in parameter_generator
        job_history, verbose_abbreviated_path):
      File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 4945, in job_needs_to_run
        check_input_files_exist(*params)
      File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/file_name_parameters.py", line 399, in check_input_files_exist
        "Input file '%s' does not exist" % f)
    ruffus.ruffus_exceptions.MissingInputFileError:

        No way to run job: Input file '/tmp/com.github.ocrmypdf.plmusa3s/000001.txt' does not exist

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 5887, in pipeline_run
        verbose)
      File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 5283, in fill_queue_with_job_parameters
        for params in job_parameters:
      File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 5215, in parameter_generator
        raise errt
    ruffus.ruffus_exceptions.RethrownJobError:

        Exceptions generating parameters for

        task = 'ocrmypdf.pipeline.render_hocr_page'

    Original exception:

        Exception #1
          'ruffus.ruffus_exceptions.MissingInputFileError(


            No way to run job: Input file '/tmp/com.github.ocrmypdf.plmusa3s/000001.txt' does not exist)' raised in ...
           Task = def ocrmypdf.pipeline.render_hocr_page(...):


        Traceback (most recent call last):
          File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 5118, in parameter_generator
            job_history, verbose_abbreviated_path):
          File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/task.py", line 4945, in job_needs_to_run
            check_input_files_exist(*params)
          File "/home/USER/.virtualenvs/pdfparse/lib/python3.5/site-packages/ruffus/file_name_parameters.py", line 399, in check_input_files_exist
            "Input file '%s' does not exist" % f)
        ruffus.ruffus_exceptions.MissingInputFileError:


            No way to run job: Input file '/tmp/com.github.ocrmypdf.plmusa3s/000001.txt' does not exist
```

Using:
- Python 3.5.2
- Ubuntu 16.04 (via https://github.com/Microsoft/BashOnWindows)

**Comment by jbarlow83** [comment]
What version of Tesseract is this? Is this ocrmypdf 5.0?

Does the file tessdata/configs/txt exist in Tesseract's data folder?
Alongside it there should be a files named hocr and pdf and several others?

**Comment by ahake** [comment]
I'm not sure how to check if the tessdata/configs/txt exists, since I am running in a docker container and it asks me to provide files when I invoke it, but I can confirm that I have the same issue. Is there a way to run the container without providing an input and output file?

**Comment by jbarlow83** [comment]
Are you running the jbarlow83/ocrmypdf-tess4 container or something else?

**Comment by ahake** [comment]
Just the standard `docker run -v d:\Programming\ocrmypdf:/home/docker **jbarlow83/ocrmypdf** --skip-text /d/Programming/ocrmypdf/test.pdf testout.pdf`.    
I've just tried again after rebooting. Now, I don't get those errors anymore, but it claims it can't find the file. Perhaps something on my end. Is there any logging I could get out?

**Comment by tobltobs** [comment]
I get the same error as the thread opener. I am running the standard jbarlow83/OCRmyPDF container, version 5.0. This error started popping up after the upgrade to 5. The version before was ok.

The old container version seems to be not available anymore on the docker repository so I can't downgrade anymore. Is there any other workaround available?

**Comment by jbarlow83** [comment]
I found the bug and released the fix in 5.0.1. It looks like that the particular version of tesseract in the docker image has a bug that makes it sensitive to order of parameters passed to it.

As of this writing's timestamp the jbarlow83/ocrmypdf is being rebuilt. An alternative, available immediately, is to use the alternative image jbarlow83/ocrmypdf-tess4 (which uses the Tesseract 4.0 alpha release).
